tests/test_barostat.py

```diff
@@ -397,6 +398,93 @@ def test_barostat_varying_pressure():
     assert compute_box_volume(atm_box) > ten_atm_box_vol
 
 
+# test that barostat only proposes properly re-centered coordinates
+def test_barostat_recentering_upon_acceptance():
```

Comment:
@jkausrelay  I added this extra test that verifies the barostat only proposes molecular configurations whose centroids are recentered into the homebox. lmk if this is sufficient for approval (or if you'd like more tests).